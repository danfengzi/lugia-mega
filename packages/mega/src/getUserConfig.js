/**
 * Created Date: Thursday, May 30th 2019, 3:32:32 pm
 * Author: hanjingbo@ysstech.com | jingboup@gmail.com
 * -----
 * Last Modified: Sunday, June 2nd 2019, 7:31:59 pm
 * Modified By: hanjingbo <hanjingbo@ysstech.com | jingboup@gmail.com>
 * -----
 * Copyright (c) 2019-present, #Lugia#.
 * ------------------------------------
 * JavaScript will save your soul!
 */

import { join } from 'path';
import excapeRegExp from 'lodash.escaperegexp';
import getUserConfig from '@lugia/mega-config';
import registerBabel from '@lugia/mega-utils/lib/registerBabel';

process.env.NODE_ENV = 'test';

const CONFIG_FILE_NAME = 'lugia.config.js';

export default function(cwd) {
  registerBabel({
    only: [new RegExp(excapeRegExp(join(cwd, CONFIG_FILE_NAME)))],
    babelPreset: [
      require.resolve('@lugia/babel-preset-mega'),
      {
        corejs: false,
        helpers: false,
      },
    ],
    disablePreventTest: true,
  });

  return getUserConfig({
    cwd,
    configFileName: CONFIG_FILE_NAME,
  });
}
