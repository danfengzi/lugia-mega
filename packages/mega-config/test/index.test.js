/**
 * Created Date: Thursday, March 28th 2019, 4:32:12 pm
 * Author: hanjingbo@ysstech.com | jingboup@gmail.com
 * -----
 * Last Modified: Thursday, March 28th 2019, 4:44:45 pm
 * Modified By: hanjingbo <hanjingbo@ysstech.com | jingboup@gmail.com>
 * -----
 * Copyright (c) 2019-present, #Lugia#.
 * ------------------------------------
 * JavaScript will save your soul!
 */

import { join } from 'path';
import getUserConfig, { getUserConfigPlugins } from '../lib';

describe('mega-config', () => {
  test('getUserConfigPlugins()', () => {
    expect(getUserConfigPlugins()).toMatchSnapshot();
  });

  test('getUserConfig()', () => {
    expect(
      getUserConfig({ cwd: join(__dirname, 'fixtures') }),
    ).toMatchSnapshot();
  });
});
