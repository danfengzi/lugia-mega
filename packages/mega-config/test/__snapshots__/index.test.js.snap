// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mega-config getUserConfig() 1`] = `
Object {
  "config": Object {
    "alias": Object {
      "react": "./react",
    },
  },
  "userPKG": Object {
    "name": "test",
    "version": "1.0.0",
  },
  "watch": [Function],
}
`;

exports[`mega-config getUserConfigPlugins() 1`] = `
Array [
  Object {
    "name": "alias",
    "validate": [Function],
  },
  Object {
    "name": "applyWebpack",
    "validate": [Function],
  },
  Object {
    "name": "babel",
    "validate": [Function],
  },
  Object {
    "name": "browserslist",
    "validate": [Function],
  },
  Object {
    "name": "clean",
    "validate": [Function],
  },
  Object {
    "name": "commons",
    "validate": [Function],
  },
  Object {
    "name": "copy",
    "validate": [Function],
  },
  Object {
    "name": "cssModulesExcludes",
    "validate": [Function],
  },
  Object {
    "name": "cssModulesWithAffix",
    "validate": [Function],
  },
  Object {
    "name": "define",
    "validate": [Function],
  },
  Object {
    "name": "devtool",
    "validate": [Function],
  },
  Object {
    "name": "disableBrowserSync",
    "validate": [Function],
  },
  Object {
    "name": "disableCSSModules",
    "validate": [Function],
  },
  Object {
    "name": "disableCSSSourceMap",
    "validate": [Function],
  },
  Object {
    "name": "disableDynamicImport",
    "validate": [Function],
  },
  Object {
    "name": "entry",
    "validate": [Function],
  },
  Object {
    "name": "env",
    "validate": [Function],
  },
  Object {
    "name": "es5ImcompatibleVersions",
    "validate": [Function],
  },
  Object {
    "name": "externals",
    "validate": [Function],
  },
  Object {
    "name": "extraBabelIncludes",
    "validate": [Function],
  },
  Object {
    "name": "extraBabelPlugins",
    "validate": [Function],
  },
  Object {
    "name": "extraBabelPresets",
    "validate": [Function],
  },
  Object {
    "name": "extraMega",
    "validate": [Function],
  },
  Object {
    "name": "extraPostCSSPlugins",
    "validate": [Function],
  },
  Object {
    "name": "extraResolveExtensions",
    "validate": [Function],
  },
  Object {
    "name": "extraResolveModules",
    "validate": [Function],
  },
  Object {
    "name": "hash",
    "validate": [Function],
  },
  Object {
    "name": "html",
    "validate": [Function],
  },
  Object {
    "name": "ignoreMomentLocale",
    "validate": [Function],
  },
  Object {
    "name": "manifest",
    "validate": [Function],
  },
  Object {
    "name": "openBrowser",
    "validate": [Function],
  },
  Object {
    "name": "outputPath",
    "validate": [Function],
  },
  Object {
    "name": "proxy",
    "validate": [Function],
  },
  Object {
    "name": "publicPath",
    "validate": [Function],
  },
  Object {
    "name": "sass",
    "validate": [Function],
  },
  Object {
    "name": "serviceworker",
    "validate": [Function],
  },
  Object {
    "name": "theme",
    "validate": [Function],
  },
  Object {
    "name": "typescript",
    "validate": [Function],
  },
  Object {
    "name": "urlLoaderExcludes",
    "validate": [Function],
  },
]
`;
